src/backend/distributed/transaction/backend_data.c

```diff
@@ -779,6 +782,30 @@ MarkCitusInitiatedCoordinatorBackend(void)
 	SpinLockRelease(&MyBackendData->mutex);
 }
 
+void
+AssignGlobalProcessId(void)
+{
+	MyBackendData->globalPID = (((uint64) GetLocalGroupId()) * 10000000000) + getpid();
+	if (!IsCitusInitiatedRemoteBackend())
+	{
+		return;
+	}
+
+	StringInfo applicationName = makeStringInfo();
+	appendStringInfoString(applicationName, application_name);
+
+	char *globalPIDString = &applicationName->data[6];
+	uint64 globalPID = strtoul(globalPIDString, NULL, 10);
+
+	MyBackendData->globalPID = globalPID;
```

Comment:
we should always guard accesses to `MyBackendData` via the `MyBackendData->mutex`. Please see other usages in the code